### PR TITLE
Tech/fix rule relating to react

### PR DESCRIPTION
**Pull request by camoore1979**
Fixed the following:

* Per https://github.com/typescript-eslint/typescript-eslint/blob/master/packages/eslint-plugin/docs/rules/no-use-before-define.md#how-to-use, I did the following in `common`:

```
    'no-use-before-define': 'off',
    '@typescript-eslint/no-use-before-define': ['error'],    
```

There were dozens of ignore of the `no-use-before-define` because of this (in `polaris-components`) in particular.

* Consolidated many rules from `node` and `react` to `common` because ... they were duplicated.
* Removed unneeded Github code checks (this repo has no lint or test scripts).
* Consolidated reviewers list for this code repository to keep it simple.

Tested locally against:
- [x] polaris-components
- [x] aurora-client
- [x] aurora-service
- [x] horizon-pro-ai
- [x] polaris-ui-reference

PLEASE check against a few of the repos above.

### CODE REVIEW

File: react.js [hunk 1]
```diff
@@ -24,23 +24,16 @@ module.exports = {
       sourceType: 'module'
     },
     rules: {
-      '@typescript-eslint/no-var-requires': 'off'
+      '@typescript-eslint/no-var-requires': 'off',
+      'jsx-a11y/click-events-have-key-events': 'off',
+      '@typescript-eslint/explicit-function-return-type': 'off'
```
Comment (khgriffi259) [inline react.js:29]: Do we want this off?

File: .github/auto_assign.yml [hunk 2]
```diff
@@ -6,7 +6,7 @@ addAssignees: author
 
 # A number of reviewers added to the pull request
 # Set 0 to add all the reviewers (default: 0)
-numberOfReviewers: 1
+numberOfReviewers: 2
```
Comment (tshelton5000) [inline .github/auto_assign.yml:9]: 👍

File: react.js [hunk 3]
```diff
@@ -24,23 +24,16 @@ module.exports = {
...
-    'class-methods-use-this': 'off',
-    'consistent-return': 'warn',
-    'import/extensions': [2, {
-      'json': 'always',
-    }],
     'import/no-default-export': 'error',
     'jsx-a11y/anchor-is-valid': 'off',
     'jsx-a11y/click-events-have-key-events': 'warn',
-    "react/jsx-filename-extension": [1, { "extensions": [".jsx", ".tsx"] }],
+    'react/jsx-filename-extension': [1, { 'extensions': ['.jsx', '.tsx'] }],
```
Comment (tshelton5000) [inline react.js:36]: We need a linter for our linting package ;)

File: react.js (hunk 1 above)
Comment (tshelton5000) [reply react.js:29]: I'm with @khgriffi259 on this particular rule--I would favor adding this to `common` under the overrides for test files.  Typing inline functions rarely causes problems, and it's usually less hairy than named functions or lambdas.

**Review by tshelton5000 — CHANGES_REQUESTED** [review]
This works great, @camoore1979 in both HPAPI and polaris-ui-reference.  I am only holding the approval for the question first raised by @khgriffi259 .  

Thank you for the diligence in helping keep our code standards high!

File: react.js (hunk 1 above)
Comment (camoore1979) [reply react.js:29]: Honestly, I'm not sure why these are there at all... these are overrides for *.js files (not .ts) and those shouldn't even apply at all, they are from `@typescript-eslint` so I wonder if they're leftover rules from previous problems...

File: react.js (hunk 3 above)
Comment (camoore1979) [reply react.js:36]: 😁  Yep. I thought the same funny thing... and then like, what would lint the linter that lints the linter?

File: react.js (hunk 1 above)
Comment (camoore1979) [reply react.js:28]: They are not leftover rules and they need to be maintained. For some reason, on *.js files in our React clients (such as next.config.js), these errors/warnings occur unless these rules are turned off in the overrides for *.js files. There is likely a deeper reason why this is the case, but not sure what it is, and I can't get to it right now. Adding these here is not a change to what we were doing in the client, it's simply moving the rule from the client repo to this shared package.

**Comment by camoore1979** [comment]
@tshelton5000 @khgriffi259 I've updated a few things and kicked back up for review.

**Review by tshelton5000 — APPROVED** [review]
Looks great to me, Chad.  Thanks for the update!

**Review by khgriffi259 — APPROVED** [review]
Thanks for clarifying Chad. I have approved.

**Pull request merged**